packages/flutter/lib/src/material/slider.dart

```diff
@@ -802,7 +802,9 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
     // (which can be defined by activeColor) if the
     // RectangularSliderValueIndicatorShape is used. In all other cases, the
     // value indicator is assumed to be the same as the active color.
-    final SliderComponentShape valueIndicatorShape = sliderTheme.valueIndicatorShape ?? defaultValueIndicatorShape;
+    final SliderComponentShape valueIndicatorShape = widget.label == null
+        ? SliderComponentShape.noThumb
+        : sliderTheme.valueIndicatorShape ?? defaultValueIndicatorShape;
```

Comment:
```suggestion
    final SliderComponentShape valueIndicatorShape = sliderTheme.valueIndicatorShape ?? defaultValueIndicatorShape;
```

Let's revert this. Instead of changing the default we could skip the drawing value indicator itself.

Reply:
Apply this fix in the `_RenderSlider` class

```dart
      if (showValueIndicator) {
        _state.paintValueIndicator = (PaintingContext context, Offset offset) {
          if (attached) {
          if (attached && _labelPainter.text != null) {
```